webapp/i18n/en.json

```diff
@@ -85,11 +85,11 @@
   "ShareBoard.copiedLink": "Copied!",
   "ShareBoard.copyLink": "Copy link",
   "ShareBoard.regenerateToken": "Regenerate token",
-  "ShareBoard.share": "Publish to web and share this board to anyone",
+  "ShareBoard.share": "Publish and share this board with anyone who has the link.",
```

Comment:
Overall, adding the proper punctuation is a great improvement. However, there may be a few cases we might want to revisit later. E.g. this text is the label for a switch in the Share Dialog:
<img width="400" alt="Share board text" src="https://user-images.githubusercontent.com/46905241/132905343-ab191c8c-58ab-47ce-ab82-0018f8595d65.png">

There are other places where switch labels don't have periods at the end, e.g. the Random icons switch:
<img width="252" alt="Random icons text" src="https://user-images.githubusercontent.com/46905241/132905453-dc5e7183-c3ee-4326-b7c0-3220e9f11072.png">

It's not a huge deal either way, and we can iterate on it. Maybe something to consider for UX guidelines if not already? cc @asaadmahmood.

Reply:
Thanks, @chenilim! I actually didn't consider the buttons - any text on those should not get punctuation. And menu items don't get punctuation either. I'm giving this another review to ensure this is adhered to. 

I've added this to the UX Guidelines update list, thank you!